webapp/src/ts/services/language.service.ts

```diff
@@ -0,0 +1,161 @@
+import {Injectable} from '@angular/core';
+import {CookieService} from "ngx-cookie";
+import {TranslateService} from "@ngx-translate/core";
+import * as moment from 'moment';
+import {SettingsService} from './settings.service';
+import {UserSettingsService} from "./user-settings.service";
+
+const LOCALE_COOKIE_KEY = 'locale';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SetLanguageService {
+
+  constructor(
+    private cookieService: CookieService,
+    private translateService: TranslateService
+  ) { }
+
+  /**
+   * Set the language for the current session.
+   * @param langCode the language code, eg. 'en', 'es' ...
+   */
+  setLanguageCookie(langCode: string): string {
+    const nextYear = new Date();
```

Comment:
You're already importing moment. So maybe use moment instead of standard date?

Reply:
Removed because now the new cooki manager takes care of the date.